test/unit/org/apache/cassandra/tools/nodetool/stats/NodetoolTableStatsTest.java

```diff
@@ -0,0 +1,237 @@
+                        "            tables to be shown\n" + 
+                        "\n" + 
+                        "        -u <username>, --username <username>\n" + 
+                        "            Remote jmx agent username\n" + 
+                        "\n" + 
+                        "        --\n" + 
+                        "            This option can be used to separate command-line options from the\n" + 
+                        "            list of argument, (useful when arguments might be mistaken for\n" + 
+                        "            command-line options\n" + 
+                        "\n" + 
+                        "        [<keyspace.table>...]\n" + 
+                        "            List of tables (or keyspace) names\n" + 
+                        "\n" + 
+                        "\n";
+        Assertions.assertThat(tool.getStdout()).isEqualTo(help);
+    }
+
+    @Test
+    public void testTableStats()
+    {
+        ToolResult tool = ToolRunner.invokeNodetool("tablestats");
+
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Keyspace : system_schema"));
+        assertTrue(StringUtils.countMatches(tool.getStdout(), "Table:") > 1);
+        assertTrue(tool.getCleanedStderr().isEmpty());
+        assertEquals(0, tool.getExitCode());
+
+        tool = ToolRunner.invokeNodetool("tablestats", "system_distributed");
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Keyspace : system_distributed"));
+        assertThat(tool.getStdout(), CoreMatchers.not(CoreMatchers.containsStringIgnoringCase("Keyspace : system_schema")));
+        assertTrue(StringUtils.countMatches(tool.getStdout(), "Table:") > 1);
+        assertTrue(tool.getCleanedStderr().isEmpty());
+        assertEquals(0, tool.getExitCode());
+    }
+
+    @Test
+    public void testTableIgnoreArg()
+    {
+        ToolResult tool = ToolRunner.invokeNodetool("tablestats", "-i", "system_schema.aggregates");
+
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Keyspace : system_schema"));
+        assertThat(tool.getStdout(), CoreMatchers.not(CoreMatchers.containsStringIgnoringCase("Table: system_schema.aggregates")));
+        assertTrue(StringUtils.countMatches(tool.getStdout(), "Table:") > 1);
+        assertTrue(tool.getCleanedStderr().isEmpty());
+        assertEquals(0, tool.getExitCode());
+    }
+
+    @Test
+    public void testHumanReadableArg()
+    {
+        Arrays.asList("-H", "--human-readable").forEach(arg -> {
+            ToolResult tool = ToolRunner.invokeNodetool("tablestats", arg);
+            assertThat("Arg: [" + arg + "]", tool.getStdout(), CoreMatchers.containsStringIgnoringCase(" KiB"));
+            assertTrue("Arg: [" + arg + "]", tool.getCleanedStderr().isEmpty());
+            assertEquals("Arg: [" + arg + "]", 0, tool.getExitCode());
```

Comment:
The `"Arg: [" + arg + "]"` error message is used multiple times here, and also in `testSortArg` and `testTopArg`. I think this is not very informative, so I' either remove it or replace it by a different message for each assertion.

Reply:
This has turned out to be very useful in some cases. If you accidentally change `-sort` to `-sorta` in the code bc of a typo you need to know which of the args is failing. In commands that accept many combinations of values this has been extremely useful as well. So I am happy to change the message but I don't want to remove it. wdyt, what message would you put?

Reply:
Ok, I didn't understand that this is only used when there are multiple ways to write the option, it looks good to me as it is. By the way, what I was thinking for the messages was something like:
```java
Arrays.asList("-H", "--human-readable").forEach(option -> {
    ToolResult tool = ToolRunner.invokeNodetool("tablestats", option);
    assertThat(String.format("Unmatched pattern in stdout for option [%s], found: %s", option, tool.getStdout()),
               tool.getStdout(), CoreMatchers.containsString(" KiB"));
    assertTrue(String.format("Expected empty stderr for option [%s] but found: %s", option, tool.getCleanedStderr()),
               tool.getCleanedStderr().isEmpty());
    assertEquals(String.format("Expected exit code 0 for option [%s] but found: %s", option, tool.getExitCode()),
                 0, tool.getExitCode());
});
```

Reply:
Unless I am misundertsnading notice the usage of `assertThat` which should give you a very nice description when it fails with exactly what you're proposing. That is one of the advantages of `asserThat` afaik.

The other 2 I can change.

Reply:
You are totally right, I thought that passing the message to `assertThat` was going to override the assertion message, but that's not the case.